src/mqtt_client.rs

```diff
@@ -84,14 +85,15 @@ where
     ///
     /// # Args
     /// * `broker` - The IP address of the broker to connect to.
-    /// * `client_id` The client ID to use for communicating with the broker.
+    /// * `client_id` The client ID to use for communicating with the broker. If None, rely on the
+    ///   broker to automatically assign a client ID.
     /// * `network_stack` - The network stack to use for communication.
     ///
     /// # Returns
     /// An `MqttClient` that can be used for publishing messages and subscribing to topics.
-    pub fn new<'a>(
+    pub fn new(
         broker: IpAddr,
-        client_id: &'a str,
+        client_id: Option<&str>,
```

Comment:
Do we need `None` here or would the empty string be indication enough? With this `Some("")` and `None` have the same meaning, right?

Reply:
Removed `None` - you are correct that an empty string is the same.